aspnet/mvc/overview.rst

```diff
@@ -1,8 +1,313 @@
-.. include:: /../common/stub-topic.txt
+:version: 1.0.0
 
-|stub-icon| Overview of ASP.NET MVC
-===================================
+Overview of ASP.NET Core MVC
+============================
 
-.. include:: /../common/stub-notice.txt
+By `Steve Smith`_
 
-.. _issue: https://github.com/aspnet/Docs/issues/113
+ASP.NET Core MVC is a rich framework for building web apps and APIs using the Model-View-Controller design pattern.
+
+.. contents:: Sections:
+  :local:
+  :depth: 1
+
+What is the MVC pattern?
+------------------------
+
+The Model-View-Controller (MVC) architectural pattern separates an application into three main groups of components: Models, Views, and Controllers. This pattern helps to achieve `separation of concerns <http://deviq.com/separation-of-concerns/>`_. Using this pattern, user requests are routed to a Controller which is responsible for working with the Model to perform user actions and/or retrieve results of queries. The Controller chooses the View to display to the user, and provides it with any Model data it requires.
+
+The following diagram shows the three main components and which ones reference the others:
+
+.. image:: overview/_static/mvc.png
+
+This delineation of responsibilities helps you scale the application in terms of complexity because it’s much easier to code, debug, and test something (model, view, or controller) that has a single job rather than something that performs multiple tasks.
+
+Kinds of Models
+^^^^^^^^^^^^^^^
+
+In simple applications, there may be just one kind of model class that is used by persistence, presentation, and any business logic. However, frequently this kind of one-size-fits-all approach doesn't scale to complex applications. In that case, it may make sense to have different model types (classes) with different responsibilities. Some MVC applications will include some combination of the following types of model objects.
```

Comment:
I don't like that that text assumes there's a database involved. There are plenty of apps for which this isn't the case. Data access logic isn't part of the model (in my way of thinking) any more than HttpClient is part of a model that happens to use web APIs for persistence. That's just plumbing/infrastructure code. I'd rather not talk about this or that method of persistence as part of the definition for the model (except perhaps in noting that you may have a separate persistence model).

One of the problems with a lot of apps that are modeled after typical Microsoft samples is that they don't have sufficient separation between the model and the database, which results in high coupling and low testability and maintainability. That's what I'm trying to avoid here, if possible.